pages/docs/styled-system/component-hooks/use-radio-group.mdx

```diff
@@ -0,0 +1,92 @@
+
+## Return value
+
+The `useRadioGroup` hook returns following props
+
+| Name            | Type                                | Description                                                                    |
+| --------------- | ----------------------------------- | ------------------------------------------------------------------------------ |
+| `value`         | `StringOrNumber[]`                  | The value of radio group.                                                      |
+| `name`          | `string`                            | The name of the radio options. All radio options must use the same name.       |
+| `ref`           | `any`                               | The ref of the radio group.                                                    |
+| `isDisabled`    | `boolean`                           | A utility to manage disabled state.                                            |
+| `isFocusable`   | `boolean`                           | A utility to manage focused state.                                             |
+| `focus`         | `() => void`                        | A utility function to set the focus on the first enabled radio.                |
+| `onChange`      | `(nextValue: EventOrValue) => void` | The onChange handler for the radio group.                                      |
+| `setValue`      | `(state: StringOrNumber[]) => void` | A function to set the value of the radio group.                                |
+| `getRadioProps` | `(props?: Dict) => Dict`            | A function that takes root props and handles changes for the radio group       |
+| `getRootProps`  | `(props?: Dict) => Dict`            | A function that takes radio root props and handles changes for the radio group |
+| `htmlProps`     | `{}`                                | A function that takes radio root props and handles changes for the radio group |
+
+## Usage
+
+```jsx
+function Example() {
+  function CustomRadio(props) {
+    const { image, ...radioProps } = props
+    const { state, getInputProps, getCheckboxProps, htmlProps, getLabelProps } =
+      useRadio(radioProps)
+
+    return (
+      <chakra.label {...htmlProps} cursor='pointer'>
+        <input {...getInputProps({})} hidden />
+        <Box
+          {...getCheckboxProps()}
+          bg={state.isChecked ? 'green.200' : 'transparent'}
+          w={12}
+          p={1}
+          rounded='full'
+        >
+          <Image src={image} rounded='full' {...getLabelProps()} />
+        </Box>
+      </chakra.label>
+    )
+  }
+
+  const avatars = [
+    { name: 'Kat', image: 'https://randomuser.me/api/portraits/women/44.jpg' },
+    { name: 'Kevin', image: 'https://randomuser.me/api/portraits/men/86.jpg' },
+    { name: 'Andy', image: 'https://randomuser.me/api/portraits/men/29.jpg' },
+    { name: 'Jess', image: 'https://randomuser.me/api/portraits/women/95.jpg' },
+  ]
+
+  const { value, getRadioProps, getRootProps } = useRadioGroup({
+    defaultValue: 'Kevin',
+  })
```

Comment:
Let's add the `onChange` handler here, just to demonstrate how users can react on the changes.

Reply:
@nikolovlazar I've updated accordingly with a small example :)